modules/migrate/application/clicommands/PreferencesCommand.php

```diff
@@ -102,4 +107,175 @@ public function indexAction()
+
+            if (! $hostActions->isEmpty()) {
+                $this->migrateNavigationItems($hostActions, $directory . '/icingadb-host-actions.ini', $rc);
+            }
+
+            if (! $serviceActions->isEmpty()) {
+                $this->migrateNavigationItems($serviceActions, $directory . '/icingadb-service-actions.ini', $rc);
+            }
+        }
+
+        // Start migrating shared navigation items
+        $hostActions = $this->readFromIni($sharedNavigation . '/host-actions.ini', $rc);
+        $serviceActions = $this->readFromIni($sharedNavigation . '/service-actions.ini', $rc);
+
+        Logger::info('Migrating shared monitoring navigation items to the Icinga DB Web actions');
+
+        if (! $hostActions->isEmpty()) {
+            $this->migrateNavigationItems($hostActions, $sharedNavigation . '/icingadb-host-actions.ini', $rc);
+        }
+
+        if (! $serviceActions->isEmpty()) {
+            $this->migrateNavigationItems($serviceActions, $sharedNavigation . '/icingadb-service-actions.ini', $rc);
+        }
+
+        if ($rc > 0) {
+            Logger::error('Failed to migrate some monitoring navigation items');
+            exit($rc);
+        }
+
+        Logger::info('Successfully migrated all local user monitoring navigation items');
+    }
+
+    /**
+     * @param Config $config
+     * @param string $path
+     * @param int    $rc
+     */
+    private function migrateNavigationItems($config, $path, &$rc)
+    {
+        $deleteLegacyFiles = $this->params->get('delete');
+        $newConfig = $this->readFromIni($path, $rc);
+        $counter = 1;
+
+        /** @var ConfigObject $configObject */
+        foreach ($config->getConfigObject() as $configObject) {
+            // Change the config type from "host-action" to icingadb's new action
+            if (strpos($path, 'icingadb-host-actions') !== false) {
+                $configObject->type = 'icingadb-host-action';
+            } else {
+                $configObject->type = 'icingadb-service-action';
+            }
+
+            $urlString = $configObject->get('url');
+            if ($urlString !== null) {
+                $url = Url::fromPath($urlString, [], new Request());
+
+                try {
+                    $urlString = UrlMigrator::transformUrl($url)->getAbsoluteUrl();
+                    $configObject->url = rawurldecode($urlString);
+                } catch (\Exception $err) {
+                    Logger::error($err->getMessage());
+                }
+            }
+
+            $legacyFilter = $configObject->get('filter');
+            if ($legacyFilter !== null) {
+                $filter = QueryString::parse($legacyFilter);
+
+                try {
+                    $filter = UrlMigrator::transformFilter($filter);
+                    $configObject->filter = rawurldecode(QueryString::render($filter));
+                } catch (\Exception $err) {
+                    Logger::error($err->getMessage());
```

Comment:
This won't throw a `InvalidArgumentException`, so please remove the try..catch block.